transport/grpc/config_test.go

```diff
@@ -353,12 +375,67 @@ func TestTransportSpec(t *testing.T) {
 					dialer, ok := single.Transport().(*Dialer)
 					require.True(t, ok, "expected *Dialer, got %T", single.Transport())
 					assert.Equal(t, wantOutbound.TLS, dialer.options.creds != nil)
+					if wantOutbound.WantCustomContextDialer {
+						assert.NotNil(t, dialer.options.contextDialer, "expected custom context dialer")
+					}
 				}
 			}
 		})
 	}
 }
 
+func TestContextDialerOptionUsage(t *testing.T) {
+	type attrs map[string]interface{}
+
+	lis, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	defer lis.Close()
+	server := grpc.NewServer()
+	defer server.Stop()
+	go func() {
+		require.NoError(t, server.Serve(lis))
+	}()
+
+	dialContextInvoked := int32(0)
+	dialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		atomic.AddInt32(&dialContextInvoked, 1)
+		return (&net.Dialer{}).DialContext(ctx, "tcp", addr)
+	}
+	configurator := yarpcconfig.New()
+	err = configurator.RegisterTransport(TransportSpec(ContextDialer(dialer)))
+	require.NoError(t, err)
+	cfgData := attrs{
+		"outbounds": attrs{
+			"myservice": attrs{
+				TransportName: attrs{"address": lis.Addr().String()},
+			},
+		},
+	}
+	cfg, err := configurator.LoadConfig("foo", cfgData)
```

Comment:
is using the configurator the recommended way to create an outbound? (fine if it is, it just seems a little verbose)

Reply:
Could be created by invoked create outbound method of TransportSpec, but it requires configurator arguments which would make it even more verbose
https://github.com/yarpc/yarpc-go/blob/4afe40930220508a6cce0af1242c959b5028b276/transport/grpc/config.go#L257